diff -Naur a52dec_original/Makefile.am a52dec_patched/Makefile.am
--- a52dec_original/Makefile.am	2001-12-17 08:23:45 +0900
+++ a52dec_patched/Makefile.am	2011-05-21 13:23:47 +0900
@@ -1,3 +1,5 @@
 SUBDIRS = autotools include doc liba52 libao src vc++ test
 
+ACLOCAL_AMFLAGS = -I m4
+
 EXTRA_DIST = HISTORY bootstrap
diff -Naur a52dec_original/configure.in a52dec_patched/configure.in
--- a52dec_original/configure.in	2002-07-28 12:50:20 +0900
+++ a52dec_patched/configure.in	2011-05-21 13:23:02 +0900
@@ -2,6 +2,7 @@
 AC_PREREQ(2.52)
 AC_INIT
 AC_CONFIG_SRCDIR([src/a52dec.c])
+AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_AUX_DIR(autotools)
 AC_CONFIG_FILES([Makefile autotools/Makefile include/Makefile test/Makefile
     doc/Makefile src/Makefile liba52/Makefile libao/Makefile vc++/Makefile])
@@ -21,7 +22,7 @@
 
     dnl -O3
     changequote(<<,>>)
-    OPT_CFLAGS=`echo "$CFLAGS"|sed "s/-O[0-9]*//g"`
+    OPT_CFLAGS=`echo "$CFLAGS"|sed "s/-O[0-9s]*//g"`
     changequote([,])
     OPT_CFLAGS="$OPT_CFLAGS -O3"
     AC_TRY_CFLAGS([$OPT_CFLAGS],[CFLAGS=$OPT_CFLAGS])
@@ -34,11 +35,11 @@
     case "$host" in
     i?86-* | k?-*)
 	case "$host" in
-	i386-*) OPT_CFLAGS="$CFLAGS -mcpu=i386";;
-	i486-*) OPT_CFLAGS="$CFLAGS -mcpu=i486";;
-	i586-*) OPT_CFLAGS="$CFLAGS -mcpu=pentium";;
-	i686-*) OPT_CFLAGS="$CFLAGS -mcpu=pentiumpro";;
-	k6-*)   OPT_CFLAGS="$CFLAGS -mcpu=k6";;
+	i386-*) OPT_CFLAGS="-march=i386 $CFLAGS";;
+	i486-*) OPT_CFLAGS="-march=i486 $CFLAGS";;
+	i586-*) OPT_CFLAGS="-march=i586 $CFLAGS";;
+	i686-*) OPT_CFLAGS="-march=i686 $CFLAGS";;
+	k6-*)   OPT_CFLAGS="-march=k6 $CFLAGS";;
 	esac
 	AC_TRY_CFLAGS([$OPT_CFLAGS],[CFLAGS=$OPT_CFLAGS]);;
     sparc-* | sparc64-*)
